google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -0,0 +1,260 @@
+import com.google.ads.googleads.v3.services.SearchGoogleAdsRequest;
+import com.google.common.base.Strings;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Gets the account hierarchy of the specified manager account. If you don't specify manager
+ * customer ID, the example will instead print the hierarchies of all accessible customer accounts
+ * for your authenticated Google account. Note that if the list of accessible customers for your
+ * authenticated Google account includes accounts within the same hierarchy, this example will
+ * retrieve and print the overlapping portions of the hierarchy for each accessible customer.
+ */
+public class GetAccountHierarchy {
+
+  private static final int PAGE_SIZE = 1_000;
+
+  private static class GetAccountHierarchyParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.MANAGER_ID)
+    private Long managerId;
+  }
+
+  public static void main(String[] args) {
+    GetAccountHierarchyParams params = new GetAccountHierarchyParams();
+    if (!params.parseArguments(args)) {
+
+      // Optional: You may pass the managerId on the command line or specify a managerId here. If
+      // neither are set, a null value will be passed to the run example method, and the example
+      // will print the hierarchies of all accessible customer IDs.
+      // params.managerId = Long.parseLong("INSERT_MANAGER_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new GetAccountHierarchy().runExample(googleAdsClient, params.managerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the examples.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param managerId managerCustomerId the customer ID.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, Long managerId) {
```

Comment:
Would suggest testing this on a large and complex manager hierarchy.

Reply:
I have test against a hierarchy with multiple levels of managers containing a mix of clients and managers in each level.